stdlib/LinearAlgebra/src/symmetriceigen.jl

```diff
@@ -165,25 +165,49 @@ function eigen!(A::Hermitian{T,S}, B::Hermitian{T,S}; sortby::Union{Function,Not
     GeneralizedEigen(sorteig!(vals, vecs, sortby)...)
 end
 function eigen!(A::RealHermSymComplexHerm{T,<:StridedMatrix}, B::AbstractMatrix{T}; sortby::Union{Function,Nothing}=nothing) where {T<:Number}
-    return _choleigen!(A, B, sortby)
+    return eigen!(Matrix{T}(A), B; sortby)
 end
 function eigen!(A::StridedMatrix{T}, B::Union{RealHermSymComplexHerm{T},Diagonal{T}}; sortby::Union{Function,Nothing}=nothing) where {T<:Number}
-    return _choleigen!(A, B, sortby)
+    return eigen!(A, Matrix{T}(B); sortby) ;
 end
-function _choleigen!(A, B, sortby)
-    U = cholesky(B).U
-    vals, w = eigen!(UtiAUi!(A, U))
-    vecs = U \ w
+
+function eigen(A::AbstractMatrix{T}, C::Cholesky{T, <:AbstractMatrix}; sortby::Union{Function,Nothing}=nothing) where {T<:Number}
+    eigen!(copy(A), C; sortby)
+end
+function eigen!(A::AbstractMatrix{T}, C::Cholesky{T, <:AbstractMatrix}; sortby::Union{Function,Nothing}=nothing) where {T<:Number}
+    # Cholesky decomposition based eigenvalues and eigenvectors
+    vals, w = eigen!(UtiAUi!(A, C.U))
+    vecs = C.U \ w
+    GeneralizedEigen(sorteig!(vals, vecs, sortby)...)
+end
+
+function eigen(A::AbstractMatrix{T}, B::BunchKaufman{T,<:StridedMatrix}; sortby::Union{Function,Nothing}=nothing) where {T<:Number}
+    eigen!(copy(A), copy(B); sortby)
```

Comment:
It seems like we don't need the copy of `B`? Whenever you need a copy, you ask for it explicitly below (in the `lu!` call). If that's the case, we could also get rid of the `copy` method in `bunchkaufman.jl`.

Reply:
[ACCEPTED] The change is in the commit: https://github.com/aravindh-krishnamoorthy/julia/commit/d8561f157162bf38859722c6fecc1c29a32d6c96